WordPress/gutenberg
# RNMobile: image block vol 03 (sizes)

## Description


This PR adds the ability to the image block of interpreting size changes made by the user.

I added a temporal size setting UI for testing purposes, but this should be replaced later with the definitive block settings UI.

I keep following closely the logic used on the web version, so (hopefully) anybody can later on check it out and understand it easily.

The visual resizing of the image currently doesn't behave exactly as on the web version, but this PR is intended to add the basics of resizing. Layer on we can decide how do we want this to behave exactly on mobile.

This is not supposed to handle image updates (change of images) either.

## Screenshots 
![image_block_resizing](https://user-images.githubusercontent.com/9772967/45636846-3e155b00-ba7f-11e8-92a2-b3654583b9d8.jpg)

**To test:**
- Checkout the [mobile-gutenberg PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/147) for this feature.
- Run the project.
- Check that the image is displaying.
- Select the image block.
- Check that it shows the size controls at the top with the default image size.
- Change the size of the image using the controls.
- Check that the image resizes accordingly.
- Check that the image never grows bigger than the block canvas.
- Switch to HTML output view.
- Check that the size changes are reflected on the HTML code (as shown on the screenshot)
